src/com/owncloud/android/ui/adapter/X509CertificateViewAdapter.java

```diff
@@ -63,25 +66,53 @@ public void updateCertificateView(View dialogView) {
         }
     }
 
+    private byte[] getDigest(String algorithm, byte[] message) {
+        MessageDigest md = null;
+
+        try {
+            md = MessageDigest.getInstance(algorithm);
+        } catch (NoSuchAlgorithmException e) {
+            // e.printStackTrace();
+            return null;
+        }
+        md.reset();
+        return md.digest(message);
+    }
+
     private void showSignature(View dialogView) {
-        TextView sigView = ((TextView)dialogView.findViewById(R.id.value_signature));
+        byte[] cert = null;
+        String newLine = System.getProperty("line.separator");
+
+        try {
+            cert = mCertificate.getEncoded();
+        } catch (CertificateEncodingException e) {
+            e.printStackTrace();
+        }
+
+        TextView certFingerprintView = ((TextView)dialogView.findViewById(R.id.value_certificate_fingerprint));
         TextView algorithmView = ((TextView)dialogView.findViewById(R.id.value_signature_algorithm));
-        sigView.setText(getHex(mCertificate.getSignature()));
+        certFingerprintView.setText("SHA-256:" + newLine
```

Comment:
If one of the digest algorithms doesn't exist, this will result in a text like

"SHA-1: NULL